pkg/network/ebpf/c/protocols/tls/java-tls-erpc.h

```diff
@@ -0,0 +1,108 @@
+#ifndef _JAVA_TLS_ERPC_H
+#define _JAVA_TLS_ERPC_H
+
+#include "bpf_helpers.h"
+#include "tracer.h"
+#include "tags-types.h"
+#include "https.h"
+#include "port_range.h"
+
+#define USM_IOCTL_ID 0xda7ad09
+
+enum erpc_message_type {
+    REQUEST,
+    CLOSE_CONNECTION
+};
+
+
+/*
+  handle_request pseudo format of *data that contain the http payload
+
+  struct {
+      u32 len;
+      u8 data[len];
+  }
+*/
+int __attribute__((always_inline)) handle_request(struct pt_regs *ctx, conn_tuple_t* connection, void *data) {
+    const bool val = true;
+    u32 bytes_read = 0;
+
+    // read the actual length of the message (limited by HTTP_BUFFER_SIZE)
+    if (0 != bpf_probe_read_user(&bytes_read, sizeof(bytes_read), data)){
+#ifdef DEBUG
+        u64 pid_tgid = bpf_get_current_pid_tgid();
+        u64 pid = pid_tgid >> 32;
+        log_debug("[java-tls-handle_request] failed reading message length location for pid %d\n", pid);
+#endif
+        return 1;
+    }
+    // register the connection in our map
+    bpf_map_update_with_telemetry(java_tls_connections, connection, &val, BPF_ANY);
+    log_debug("[java-tls-handle_request] handling tls request of size: %d\n", bytes_read);
+    https_process(connection, data+sizeof(bytes_read), bytes_read, JAVA_TLS);
+    http_flush_batch(ctx);
+    return 0;
+}
+
+void __attribute__((always_inline)) handle_close_connection(conn_tuple_t* connection) {
+    void *exists = bpf_map_lookup_elem(&java_tls_connections, connection);
+    // if the connection exists in our map, finalize it and remove from the map
+    // otherwise just ignore
+    if (exists != NULL){
```

Comment:
I think we should use LRU map for java_tls_connections, as in case we are "missing" the close connection somehow

Reply:
agreed, go_tls need too (will be in a different PR)

Reply:
more generally all maps with conn_tuple_t should be LRU and rewrite entries
that the kind of checker I will add